src/transaction/transaction_manager.cpp

```diff
@@ -177,11 +178,10 @@ timestamp_t TransactionManager::Abort(TransactionContext *const txn) {
 
   // We hand off txn to GC, however, it won't be GC'd until the LogManager marks it as serialized
   if (gc_enabled_) {
-    common::SpinLatch::ScopedSpinLatch guard(&timestamp_manager_->curr_running_txns_latch_);
     // It is not necessary to have to GC process read-only transactions, but it's probably faster to call free off
```

Comment:
I'm not convinced this is true now that we're taking a latch that other txns will be trying to get in the middle of query execution, rather than just at commit.. I would change this to a TODO: that says "evaluate performance of deleting read-only transactions on the critical path".